x-pack/plugins/elastic_assistant/server/routes/helpers.ts

```diff
@@ -87,3 +91,69 @@ export const hasAIAssistantLicense = (license: ILicense): boolean =>
 
 export const UPGRADE_LICENSE_MESSAGE =
   'Your license does not support AI Assistant. Please upgrade your license.';
+
+interface PerformChecksParams {
+  authenticatedUser?: boolean;
+  capability?: AssistantFeatureKey;
+  context: AwaitedProperties<
+    Pick<ElasticAssistantRequestHandlerContext, 'elasticAssistant' | 'licensing' | 'core'>
+  >;
+  license?: boolean;
+  request: KibanaRequest;
+  response: KibanaResponseFactory;
+}
+
+/**
+ * Helper to perform checks for authenticated user, capability, and license. Perform all or one
+ * of the checks by providing relevant optional params. Check order is license, authenticated user,
+ * then capability.
```

Comment:
Noticed our routes are starting to bloat with common checks like license, authUser or if a feature flag is enabled, so I made this helper to do any or all of those checks in a single block.